keyring_darwin.go

```diff
@@ -48,15 +51,22 @@ func (k macOSXKeychain) Get(service, username string) (string, error) {
 	// if the added secret has multiple lines, osx will hex encode it
 	trimStr := strings.TrimSpace(string(out[:]))
 	dec, err := hex.DecodeString(trimStr)
-	// if there was an error hex decoding the string, assume it's not encoded
-	if err != nil {
-		return fmt.Sprintf("%s", trimStr), nil
+	// if there was no error hex decoding the string and it has the well-known prefix, assume it's encoded
```

Comment:
I would check for the prefix first and only hex decode it in that case. Meaning that when we `Set` we should do the hex encoding and not rely on what osx will do.

Reply:
Hmm, it's already doing the hex encoding during the Set, assuming it sees a multiline string.

Should I perhaps add the prefix outside of the encoded string so that `Get` can check for it without needing to decode? E.g. if a multiline string is `Set` then the keychain would get something like:

```
go-keyring-encoded:746869732070617373776f72640a686173206d756c7469706c650a6c696e657320616e642077696c6c2062650a656e636f64656420627920736f6d65206b657972696e6720696d706c656d656e74696174696f6e730a6c696b65206f7378
```

Reply:
Instead of relying on osx doing the hex encoding I would do it manually in `Set` so we control it. Basically if the password has a newline, hex encode it and add the prefix. I think the code would also be easier to understand that way because there are no weird side effects from osx.

Reply:
Sorry, now I'm a little confused. Isn't that what I'm doing in the `Set` already?

```golang
	if strings.ContainsRune(password, '\n') {
		password = hex.EncodeToString([]byte(encodingPrefix + password))
	}
```

Although (if I interpreted correctly) I think I like your idea of placing the prefix outside of the encoded string to make it easier for `Get` to find. If that's where you're headed then I would do this:

```diff
- password = hex.EncodeToString([]byte(encodingPrefix + password))
+ password = encodingPrefix + hex.EncodeToString([]byte(password))
```

Reply:
FYI I pushed the change that places the encodingPrefix outside of the encoded string. Hopefully this is what you were looking for! :)

Reply:
Yes, that is what I mean. I think this makes the code much simpler to understand!